src/Options.purs

```diff
@@ -0,0 +1,74 @@
+module Options
+    ( OptionSpecifications
+    , OptionSpecification
+    , OptionValues
+    , OptionValue
+    , Option
+    , OptionValueExtractor
+    , booleanOption
+    , makeOptionValues
+    , lookupOptionValue
+    ) where
+
+import Prelude
+
+import Data.Maybe (Maybe(..), fromJust)
+import Data.StrMap (StrMap)
+import Data.StrMap as SM
+import Data.Tuple (Tuple(..))
+import Data.String as String
+import Partial.Unsafe (unsafePartial)
+
+-- FIXME: all of this should be strongly typed and not require
+-- unsafePartial.
+
+data OptionValue
+    = BooleanValue Boolean
+
+type OptionSpecification =
+    { name :: String
+    , description :: String
+    , typeLabel :: String
+    , default :: OptionValue
+    }
+
+type OptionSpecifications = Array OptionSpecification
```

Comment:
I think you can define `OptionSpecifications` as an Array of records that contain `specification :: OptionSpecification`:

```purs
type OptionSpecifications = Array { specification :: OptionSpecification | r }
```

I'm not sure if you have to quantify the `r` (you might need a `forall r.`) but this should let renderers do:

```purs
, options: [listOption]
```

without having to add `.specification`.

Reply:
I can't quite get this to work but I think it's possible.

Reply:
I don't understand how that works.